### PR TITLE
Testhostflatten

### PR DESCRIPTION
**Pull request by acesiddhu**
1) Flatten testhost to root folder
2) added Qtool package which make
a) mstest with legacy to work
b) Webtest
c) generic test
d) etc to work

Qtools version needs to bump with a minor bug fix
End to end testing with VS is pending. ran test.cmd

### CODE REVIEW

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 1]
```diff
@@ -144,8 +144,15 @@ public TestHostConnectionInfo GetTestHostConnectionInfo()
             var currentWorkingDirectory = Path.Combine(Path.GetDirectoryName(typeof(DefaultTestHostManager).GetTypeInfo().Assembly.Location), "..//");
             var argumentsString = " " + connectionInfo.ToCommandLineOptions();
 
-            // "TestHost" is the name of the folder which contain Full CLR built testhost package assemblies.
-            testHostProcessName = Path.Combine("TestHost", testHostProcessName);
+            // check in current location for testhost exe
+            var testhostProcessPath = Path.Combine(currentWorkingDirectory, testHostProcessName);
+
+            if (!File.Exists(testhostProcessPath))
```
Comment (smadala) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: Similar check need to do before launching `QTAgent.exe` to support `dotnet test`.

File: scripts/build.ps1 [hunk 2]
```diff
@@ -344,12 +349,12 @@ function Create-VsixPackage
     $vsixSourceDir = Join-Path $env:TP_ROOT_DIR "src\package\VSIXProject"
     $vsixProjectDir = Join-Path $env:TP_OUT_DIR "$TPB_Configuration\VSIX"
     $packageDir = Get-FullCLRPackageDirectory
-    $testhostPackageDir = Join-Path $packageDir "TestHost"
+    $testhostPackageDir = Join-Path $packageDir ""
```
Comment (smadala) [inline scripts/build.ps1]: Should we remove this and use `$packageDir` path directly?

File: scripts/build.ps1 [hunk 3]
```diff
@@ -233,11 +233,16 @@ function Publish-Package
...
+    Publish-PackageInternal $testHostx86Project $TPB_TargetFramework $testhostFullPackageDir    
+
+    # Copy MSTestV1 dependencies to testhost folder
+    $mstestDir = Join-Path $env:TP_PACKAGES_DIR "Microsoft.VisualStudio.QualityTools\15.5.0-preview-977265\contentFiles\any\any"
+    Copy-Item -Recurse $mstestDir\* $testhostFullPackageDir -Force
 
     # Copy over the Full CLR built testhost package assemblies to the Core CLR and Full CLR package folder.
-    $netFull_Dir = "TestHost"
-    $fullDestDir = Join-Path $coreCLR20PackageDir $netFull_Dir
+    $netFull_Dir = ""
```
Comment (smadala) [inline scripts/build.ps1]: Directly copy to `fullCLRPackageDir`.

**Review by smadala — COMMENTED** [review]
Please add acceptance tests for all new test adapters.
👍

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs (hunk 1 above)
Comment (acesiddhu) [reply src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: as discussed this is not needed. but I will confirm this once

Comment (acesiddhu) [reply src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: as dicussed with Arun offline we don't want to support Qtagent scenarios with dotnet test command line

File: scripts/build.ps1 [hunk 4]
```diff
@@ -233,20 +233,19 @@ function Publish-Package
...
     Copy-Item $testhostFullPackageDir\* $fullDestDir -Force -recurse
 
     # Copy over the Full CLR built datacollector package assemblies to the Core CLR package folder along with testhost
     Publish-PackageInternal $dataCollectorProject $TPB_TargetFramework $fullDestDir
-
-    $fullDestDir = Join-Path $fullCLRPackageDir $netFull_Dir
-    New-Item -ItemType directory -Path $fullDestDir -Force | Out-Null
-    Copy-Item $testhostFullPackageDir\* $fullDestDir -Force -recurse
+    
+    New-Item -ItemType directory -Path $fullCLRPackageDir -Force | Out-Null
```
Comment (Faizan2304) [inline scripts/build.ps1]: $fullCLRPackageDir is already there, no need to create again.

**Review by Faizan2304 — APPROVED** [review]

**Review by codito — APPROVED** [review]

File: scripts/build.ps1 (hunk 4 above)
Comment (codito) [reply scripts/build.ps1]: Merging this PR in interest of time, will clean this one in next PR.

**Pull request merged**